quickwit-cli/src/indexing/byte_units.rs

```diff
@@ -0,0 +1,128 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use std::convert::TryFrom;
+
+/// Base 2 bytes.
+#[derive(Clone, Debug)]
+pub enum ByteUnits {
+    Bytes(u64),
+    KiB(u64),
+    MiB(u64),
+    GiB(u64),
+}
+
+impl ByteUnits {
+    pub fn bytes(&self) -> u64 {
+        match *self {
+            ByteUnits::Bytes(bytes) => bytes,
+            ByteUnits::KiB(kib) => kib * (1 << 10),
+            ByteUnits::MiB(mib) => mib * (1 << 20),
+            ByteUnits::GiB(gib) => gib * (1 << 30),
+        }
+    }
+}
+
+/// Implementation of String to ByteUnits coversion
+///
+/// The following regex defines the specification:
+/// ^[0-9]+_?(b|bytes|bytes|kb|kib|mb|mib|gb|gib)?$
+/// Examples:
+/// 2, 2b, 2byte, 2_bytes => ByteUnits::Bytes(2)
+/// 2kb, 2kib => ByteUnits::KiB(2)
+/// 2mb, 2mib => ByteUnits::MiB(2)
+/// 2gb, 2gib => ByteUnits::GiB(2)
```

Comment:
Can we replace all of the b by B in the docstring?

I know the lowercase normalization makes it so that both are correct, but kb usually   means kilobits.
We might want to avoid confusing the user.

Ideally we should explicitly explain that in the comment.

"This function does not interpret lowercase b as meaning 'bit'. It interprets 'b' regardless of its case, as Byte."

Reply:
I just found out this was a waste of energy. it exists in this crate `byte-unit`

Reply:
it looks perfect indeed